c10/util/StringUtil.cpp

```diff
@@ -28,6 +37,94 @@ std::string ExcludeFileExtension(const std::string& file_name) {
   return file_name.substr(0, end_index);
 }
 
+// Narrows the wstr argument and then passes it to _str.
+// Assumes that the input (wide) text is encoded as UTF-16.
+template <typename WType>
+std::ostream& _strFromWide(std::ostream& ss, const WType& wideText);
+
+#ifndef _WIN32
+
+template <typename WType>
+std::ostream& _strFromWide(std::ostream& ss, const WType& wideText) {
+  std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
+  return _str(ss, converter.to_bytes(wideText));
+}
+
+#else // #ifndef _WIN32
+// The WIN32 implementation of wstring_convert leaks memory; see
+// https://github.com/microsoft/STL/issues/443
+
+template <>
```

Comment:
This feels like a huge code duplication from https://github.com/pytorch/pytorch/blob/208e64a9ba2abca63ea77b3b2dc5a2f19e4c2ac4/c10/util/Unicode.cpp#L26-L34 can perhaps template from Unicode.cpp be extended and used here?